versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java

```diff
@@ -540,6 +524,83 @@ public void upsertObjs(@Nonnull Obj[] objs) throws ObjTooLargeException {
     }
   }
 
+  @Override
+  public boolean deleteConditional(@Nonnull UpdateableObj obj) {
+    ConditionalRowMutation conditionalRowMutation =
+        mutationForConditional(obj, Mutation.create().deleteRow());
+    return backend.client().checkAndMutateRow(conditionalRowMutation);
+  }
+
+  @Override
+  public boolean updateConditional(@Nonnull UpdateableObj expected, @Nonnull UpdateableObj newValue)
+      throws ObjTooLargeException {
+    ObjId id = expected.id();
+    checkArgument(id != null && id.equals(newValue.id()));
+    checkArgument(expected.type().equals(newValue.type()));
+    checkArgument(!expected.versionToken().equals(newValue.versionToken()));
+
+    Mutation mutation = objectWriteMutation(newValue, false);
+
+    ConditionalRowMutation conditionalRowMutation = mutationForConditional(expected, mutation);
+    return backend.client().checkAndMutateRow(conditionalRowMutation);
+  }
+
+  @NotNull
+  private ConditionalRowMutation mutationForConditional(
+      @NotNull UpdateableObj obj, Mutation mutation) {
+    checkArgument(obj.id() != null, "Obj to store must have a non-null ID");
+    ByteString key = dbKey(obj.id());
+
+    Filters.ChainFilter objTypeFilter =
+        FILTERS
+            .chain()
+            .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_TYPE))
+            .filter(FILTERS.value().exactMatch(OBJ_TYPE_VALUES.get(obj.type())));
+    Filters.ChainFilter objVersionFilter =
+        FILTERS
+            .chain()
+            .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_VERS))
+            .filter(FILTERS.value().exactMatch(obj.versionToken()));
+    Filter condition =
+        FILTERS.condition(objTypeFilter).then(objVersionFilter).otherwise(FILTERS.block());
+
+    return ConditionalRowMutation.create(backend.tableObjs, key)
+        .condition(condition)
+        .then(mutation);
+  }
+
+  @NotNull
+  private Mutation objectWriteMutation(@NotNull Obj obj, boolean ignoreSoftSizeRestrictions)
+      throws ObjTooLargeException {
+    int incrementalIndexSizeLimit =
+        ignoreSoftSizeRestrictions ? Integer.MAX_VALUE : effectiveIncrementalIndexSizeLimit();
+    int indexSizeLimit =
+        ignoreSoftSizeRestrictions ? Integer.MAX_VALUE : effectiveIndexSegmentSizeLimit();
+    byte[] serialized = serializeObj(obj, incrementalIndexSizeLimit, indexSizeLimit, false);
+
+    Mutation mutation = Mutation.create();
+    objToMutation(obj, mutation::setCell, serialized);
+    return mutation;
+  }
+
+  static void objToMutation(Obj obj, ToMutation mutation, byte[] serialized) {
```

Comment:
No need for a functional interface here, you can use `MutationApi<?> mutation`.

Reply:
done + refactored